### PR TITLE
🐛  Source Amplitude: add error descriptions and fix `events` stream fail on 404

### LINKED ISSUE #7900: Source Amplitude: add better response error msg to return error 400 reason file is too big
## Tell us about the problem you're trying to solve

When users with large Amplitude data want to sync they receive a 400 error...
From Amplitude docs:
image
https://developers.amplitude.com/docs/export-api

## Describe the solution you’d like
If I receive this error maybe we need to return a better response to user the logs aren't clear.

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### LINKED ISSUE #10220: Source Amplitude: events not syncing properly
## Environment
- **Airbyte version**:  0.35.26-alpha
- **OS Version / Instance**: debian 10
- **Deployment**: docker
- **Source Connector and version**: Amplitude 0.1.4
- **Destination Connector and version**: postgres
- **Severity**: High
- **Step where error happened**: Sync job 

## Current Behavior
amplitude returns a 404 if the time range that we are trying to pull doesn't have any data, and the connector doesnt handle it and just quits.

## Expected Behavior
the connector should try the next dates until it doesnt get a 404.

## Logs
[logs-1.txt](https://github.com/airbytehq/airbyte/files/8033064/logs-1.txt)

<details>
<summary>LOG</summary> 

```
2022-02-09 13:19:32 source > Syncing stream: events 
2022-02-09 13:19:35 source > Error during syncing events stream - 404 Client Error: Not Found for url: https://amplitude.com/api/2/export?start=20210601T00&end=20210604T00
2022-02-09 13:19:35 source > Setting state of events stream to {}
2022-02-09 13:19:35 source > Read 0 records from events stream
2022-02-09 13:19:35 source > Finished syncing SourceAmplitude
2022-02-09 13:19:35 source > SourceAmplitude runtimes:
2022-02-09 13:19:35 source > Finished syncing SourceAmplitude
2022-02-09 13:19:36 INFO i.a.w.DefaultReplicationWorker(run):162 - One of source or destination thread complete. Waiting on the other.
2022-02-09 13:19:36 destination > 2022-02-09 13:19:36 INFO i.a.i.b.FailureTrackingAirbyteMessageConsumer(close):60 - {} - Airbyte message consumer: succeeded.
2022-02-09 13:19:36 destination > 2022-02-09 13:19:36 INFO i.a.i.d.b.BufferedStreamConsumer(close):200 - {} - executing on success close procedure.
2022-02-09 13:19:36 destination > 2022-02-09 13:19:36 INFO i.a.i.d.j.JdbcBufferedConsumerFactory(lambda$onCloseFunction$3):160 - {} - Finalizing tables in destination started for 4 streams
```

</details>

## Steps to Reproduce
1.create amplitude source with start date before the account was created
2.try to sync events

## Are you willing to submit a PR?
yes

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving:
https://github.com/airbytehq/airbyte/issues/10220
https://github.com/airbytehq/airbyte/issues/7900

## How
- added `error_msg_from_status` method to log the http errors correctly
- adjusted `unit_tests` to cover the case

## 🚨 User Impact 🚨
No impact expected.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [X] Documentation updated
    - [X] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by grubberr — APPROVED** [review]

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@
 - name: Amplitude
   sourceDefinitionId: fa9f58c6-2d03-4237-aaa4-07d75e0c1396
   dockerRepository: airbyte/source-amplitude
-  dockerImageTag: 0.1.4
+  dockerImageTag: 0.1.5
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-config/init/src/main/resources/seed/source_definitions.yaml:38]: ok

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 2]
```diff
@@ -27,8 +29,12 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
         return None
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        response_data = response.json()
-        yield from response_data.get(self.name, [])
+        status = response.status_code
+        if status in list(HTTP_ERROR_CODES.keys()):
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:33]: removed.

**Comment by klsoper** [comment]
Hi @bazarnov - it looks like this has broken some syncs on cloud that were previously working. 

`2022-05-11 12:14:51 [44msource[0m > Encountered an exception while reading stream events
`
`2022-05-11 12:14:53 [43mdestination[0m > java.lang.RuntimeException: com.google.cloud.bigquery.BigQueryException: 404 Not Found`

Cc @edgao because he is on call -not sure if this is a quick fix. Happy to create an alpha-beta issue to look into this further too if needed. I can slack you the workspace and connection link if you need.

**Comment by edgao** [comment]
@evantahler fyi as current oc (schedule got updated yesterday, if hubot still says I'm oc then that's a bug)

**Comment by evantahler** [comment]
Hi @klsoper let's make a new alpha-beta issue for this!  The connection between a `BigQueryException` and these Amplitude connector fixes is probably something to discuss!

**Comment by klsoper** [comment]
@evantahler Thank you- just got one going!

**Comment by mohitreddy1996** [comment]
Hi, since this seems to be the latest heavy change for Amplitude connector, looking for some help/resolution here - https://github.com/airbytehq/airbyte/issues/13057

**Pull request merged**